src/StudyEntities/Candidate/EntityType.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+/**
+ * This defines the entity type for a Candidate. It is a Value Object used
+ * to assist with validation.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  LORIS
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\StudyEntities\Candidate;
+
+/**
+ * This defines the entity type for a Candidate. It is a Value Object used
+ * to assist with validation.
+ *
+ * @category Main
+ * @package  LORIS
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class EntityType
+{
+    /* @var string */
+    private $value;
+
+    private const VALID_VALUES = array(
+                                  'Human',
+                                  'Scanner',
+                                 );
+
+    /**
+     * Creates a new EntityType. Calls validate() immediately.
+     *
+     * @param string $value The value of this EntityType.
+     *
+     * @throws \DomainException When the value is not valid
+     */
+    final public function __construct(string $value)
+    {
+        if (!$this->validate($value)) {
+            throw new \DomainException(
+                'The value is not valid. Must be one of: '
+                . implode(', ', self::VALID_VALUES)
+            );
+        }
+        $this->value = $value;
+    }
+
+    /**
+     * Ensures that EntityType is well-formed.
+     *
+     * @param string $value The value to be validated
+     *
+     * @return bool True if the value format is valid
+     */
+    private function validate(string $value): bool
```

Comment:
I think this should be public static. Nothing about the validation depends on the class state, there's no reason you shouldn't be able to validate a string without instantiating it.

Reply:
The parent class declares it as protected so I don't think it's possible to change this to public without changing other child classes of ValidatableIdentfier.

Reply:
There is no parent class. This class is the root of it's own hierarchy.

Reply:
Oops I thought i extended ValidatableIdentfier.

Do we want to move away from using the design @xlecours used in that class? It seems very good for value objects imo

Reply:
You shouldn't be able to instantiate an invalid object, but that doesn't mean you should need an object in order to validate an input.